aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -75,6 +75,26 @@ static inline std::tuple<Tensor, Tensor> _lu_det_P_diag_U(const Tensor& self) {
   return std::tuple<Tensor, Tensor>(num_exchanges.mul_(-2).add_(1), u_diagonal);
 }
 
+// Given a pivoted LU factorization A = P L U,
+// det(A) = det(P) * det(L) * det(U).
+// Since det(P) = +- 1 (even or odd permutation), and diag(L) = I, we get that
+// det(A) = ([is P odd] * -2 + 1) * prod(diag(U))
+std::tuple<Tensor, Tensor, Tensor> _det_lu_based_helper(const Tensor& self) {
+  Tensor lu, pivs, infos;
+  std::tie(lu, pivs, infos) = at::_lu_with_info(self, /*pivot=*/true, /*check_errors*/false);
+  TORCH_CHECK(infos.ge(0).all().item<uint8_t>(), "at::_det_lu_based_helper(): Invalid argument passed to LU");
```

Comment:
This triggers a synchronisation in GPU, is that right?
If so, according to [the magma docs](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__getrf__batched.html#gac5d0dfc5179e4ff7dd34971c885afb79), the PLU returns:
INFO = -i, the i-th argument had an illegal value or another error occurred, such as memory allocation failed.
Would it be then reasonable to either skip this check completely or change it by another check to avoid synchronisation?
cc @IvanYashchuk @mruberry @ngimel

Reply:
The same check is done in the previous forward for `linalg.det`. Yes, it does synchronize